test/tools/utils.js

```diff
@@ -0,0 +1,291 @@
+ * @param {Array<Number>} input
+ * @param {String} precisionType
+ * @return {(Array<Number>|Number)}
+ */
+function getPrecisionData(input, precisionType) {
+  let data;
+  const isNumber = typeof input === 'number';
+  if (isNumber) {
+    input = [input];
+  }
+
+  switch (precisionType) {
+    case 'float16':
+      data = new Float16Array(input);
+      break;
+    case 'float32':
+      data = new Float32Array(input);
+      break;
+    case 'int8':
+      data = new Int8Array(input);
+      break;
+    case 'uint8':
+      data = new Uint8Array(input);
+      break;
+    case 'int32':
+      data = new Int32Array(input);
+      break;
+    case 'uint32':
+      data = new Uint32Array(input);
+      break;
+    case 'int64':
+      data = new BigInt64Array(input.map((x) => BigInt(x)));
+      break;
+    case 'uint64':
+      data = new BigUint64Array(input.map((x) => BigInt(x)));
+      break;
+    default:
+      break;
+  }
+
+  if (isNumber) {
+    data = data[0];
+  }
+  return data;
+}
+
+/**
+ * Get converted data from given data dict with specified field and precision type.
+ * @param {Object} srcDataDict
+ * @param {String} source
+ * @param {String} precisionType
+ * @return {(Array<Number>|Number)}
+ */
+function getPrecisionDataFromDataDict(srcDataDict, source, precisionType) {
+  const feedData = srcDataDict[source];
+  return getPrecisionData(feedData, precisionType);
+}
+
+/**
+ * Get a random number by specified dataRange and dataType.
+ * @param {{min: Number, max: Number, sign: String}} dataRange
+ * @param {String} dataType
+ * @return {Number}
+ */
+function getRandom(dataRange, dataType) {
+  function getFloatRandomInclusive() {
+    // The Math. random() method returns a random floating point number between 0 (inclusive)
+    // and 1 (exclusive).
+    let value = Math.random() * 2;
+    if (value > 1) {
+      value = 1;
+    }
+    return value;
+  }
+
+  function validateMinMaxByType(min, max, type) {
+    if (type === 'int8' && (min < -128 || max > 127)) {
+      throw new Error(`The range of int8 type should be [-128, 127].`);
+    } else if (type === 'uint8' && (min < 0 || max > 255)) {
+      throw new Error(`The range of uint8 type should be [0, 255].`);
+    } else if (type === 'int32' && (min < -Math.pow(2, 31) || max > Math.pow(2, 31) - 1)) {
+      throw new Error(
+          `The range of int32 type should be [${-Math.pow(2, 31)}, ${Math.pow(2, 31) - 1}].`);
+    } else if (type === 'uint32' && (min < 0 || max > Math.pow(2, 32) - 1)) {
+      throw new Error(`The range of uint32 type should be [0, ${Math.pow(2, 32) - 1}].`);
+    } else if (type === 'float16') {
+      const fp16Max = (2- Math.pow(2, -10)) * Math.pow(2, 15);
+      if (min < -fp16Max || max > fp16Max) {
+        throw new Error(`The range of float16 type should be [${-fp16Max}, ${fp16Max}].`);
+      }
+    } else if (type === 'int64') {
+      const int64Max = 2n ** 64n - 1n;
+      if (min < -int64Max || max > int64Max) {
```

Comment:
The minimum value is one less...

`min < -int64Max - 1 || max > int64Max`

...just like int32:

`(min < -Math.pow(2, 31) || max > Math.pow(2, 31) - 1))`